docs/reserved-resource-names-model-as-enum.md

```diff
@@ -0,0 +1,26 @@
+# ReservedResourceNamesModelAsEnum
+
+## Category
+
+ARM Error
+
+## Applies to
+
+ARM OpenAPI (swagger) specs
+
+## Related ARM Guideline Code
+
+- RPC-ConstrainedCollections-V1-04
+
+## Description
+
+Service-defined (reserved) resource names must be represented as an `enum` type with `modelAsString` set to `true`, not as a static string in the path.
```

Comment:
>not as a static string in the path. [](http://example.com/codeflow?start=116&length=35)

Lets also add more rationale here.

This is to allow for expansion of the resource collection to include more service defined instances in future if necessary. Changing the representation of the path in swagger to an enum does not require you to change the implementation of the API in the service but helps with forwards compatibility.